### ISSUE TITLE
Document a possible workaround for Fedora 33

### ISSUE THREAD

**Issue by gbraad** [code-ready/crc#1624]
Ref: #1572

**Comment by gbraad** [comment]
@cfergeau did you look into the issue? Do we still need to document this?

**Comment by cfergeau** [comment]
Praveen has been testing this change:
```
# cat /etc/systemd/resolved.conf.d/crc.conf 
[Resolve]
DNS=192.168.130.11
Domains=~. testing
# resolvectl dns
Global: 192.168.130.11
Link 2 (eth0): 10.11.142.1 10.11.5.19
# resolvectl domain
Global: ~. testing
```
with the hope that it would be generic enough to work both on f33 and ubuntu. This could also be handled by following https://fedoramagazine.org/systemd-resolved-introduction-to-split-dns/ , but Praveen was worried this would not work on Ubuntu which uses systemd-networkd. https://manpages.ubuntu.com/manpages/cosmic/man1/resolvectl.1.html documents that when systemd-networkd is in use, per-interface nameservers must be set in the network interface file, while when it's not in use, it's done in a different place.

I did not test any of this myself.

**Comment by gbraad** [comment]
Do we report this currently as a known issue? An automated solution for this doesnt exist yet

**Comment by robin-owen** [comment]
@gbraad This isn't recorded as a known issue at this time. We can add it to the known issues asynchronously, however.

**Comment by cfergeau** [comment]
> Praveen has been testing this change:
> 
> ```
> # cat /etc/systemd/resolved.conf.d/crc.conf 
> [Resolve]
> DNS=192.168.130.11
> Domains=~. testing
> # resolvectl dns
> Global: 192.168.130.11
> Link 2 (eth0): 10.11.142.1 10.11.5.19
> # resolvectl domain
> Global: ~. testing
> ```
> 
> with the hope that it would be generic enough to work both on f33 and ubuntu. This could also be handled by following https://fedoramagazine.org/systemd-resolved-introduction-to-split-dns/ , but Praveen was worried this would not work on Ubuntu which uses systemd-networkd. https://manpages.ubuntu.com/manpages/cosmic/man1/resolvectl.1.html documents that when systemd-networkd is in use, per-interface nameservers must be set in the network interface file, while when it's not in use, it's done in a different place.
> 
> I did not test any of this myself.

One note about ubuntu: ubuntu desktop uses a setup which seems similar to f33 (systemd-resolved + NetworkManager, no systemd-networkd), while ubuntu server uses systemd-resolved+systemd-networkd without NetworkManager.

**Comment by hippyod** [comment]
I have a workaround for this:

* First make `libvirtd.service` a dependency of `systemd-resolved.service`.  CRC has it's own network, and that needs to be up and running for `systemd-resolved.service` to pick up.
  
    ```
    sudo vi /etc/systemd/system/dbus-org.freedesktop.resolve1.service
    ```

    and change

    ```
    After=systemd-sysusers.service systemd-networkd.service
    ```

    to

    ```
    After=systemd-sysusers.service systemd-networkd.service libvirtd.service
    ```

    Save and exit the text editor.

* Next, open the `systemd-resolved.service` configuration file, `resolved.conf`, for editing to add the CRC DNS and Domains:

    ```
    sudo vi /etc/systemd/resolved.conf
    ```

    and add

    ```
    DNS=192.168.130.11%crc#apps-crc.testing 192.168.130.11%crc#crc.testing
    Domains=~apps-crc.testing ~crc.testing
    ```

    Save and exit the text editor.

* Restart the `systemd-resolved` service:

    ```
    sudo systemctl restart systemd-resolved.service
    ```

Your system should now properly pick up the CRC network even if you reboot your machine.  Been using it for a few days now without issue.

**Comment by cfergeau** [comment]
I've been having success with https://github.com/code-ready/crc/issues/1572#issuecomment-723026134:

> I'm currently having success with adding

$ cat 10-crc
#!/bin/sh
# This is a NetworkManager dispatcher script to configure split DNS for
# the 'crc' libvirt network.
# The corresponding bridge is recreated each time the system reboots, so
# it cannot be configured permanently through NetworkManager.
# Changing DNS settings through nmcli requires the connection to go down/up,
# so we directly do the change through resolvectl

export LC_ALL=C

if [ "$1" = crc ] && [ "$2" = up ]; then
        resolvectl domain crc ~testing
        resolvectl dns crc 192.168.130.11
        resolvectl default-route crc false
fi

exit 0

in /etc/NetworkManager/dispatcher.d/10-crc and making the file executable. This will need a destroy/start of the libvirt crc network, or a reboot of the host as this is only triggered when the crc NM interface state changes to up

**Comment by hippyod** [comment]
Mr. Fergeau, that's a fabulous solution, although I changed it slightly:

```
if [ "$1" = crc ] && [ "$2" = up ]; then
    echo "checking in dispather.d $1"
    resolvectl domain crc crc.testing apps-crc.testing
    resolvectl dns crc 192.168.130.11
    resolvectl default-route crc false
fi
```

I added the echo statements so I could verify in the system logs, and as you can see I referenced the domains for fully.  I couldn't get it to work with `~testing` for some reason, so I started screwing around.  Lots of reboots seem to have it working for me now, although I am noticing that every once in a while the VM isn't coming very quickly (or at least isn't reachable quickly), and so it might have been my impatience that made me think ~testing didn't work?  Dunno.

Thanks much for what looks to be a better solution than the one we came up with.  Appreciate it.

**Comment by robin-owen** [comment]
This is now documented as a known issue based on this comment: https://github.com/code-ready/crc/issues/1572#issuecomment-728911700

@gbraad For the future, do you think we should document this in the troubleshooting section of the docs? Primarily asking because "known issues" at this time are entirely downstream, whereas the troubleshooting docs exist in this repository. If this workaround will be applicable for a while, it would make sense for us to have these docs available via GitHub Pages as well. Let me know what you think. :)

**Comment by gbraad** [comment]
@crc-team we need somehow automated or manual tests setup for Ubuntu.

**Comment by cfergeau** [comment]
This has been  fixed by #1732
